### PR TITLE
feat: Add auth pages ui

### PR DESCRIPTION
**Pull request by rishabh1403**
Add 
- Sign up Page
- Sign In Page
- Password Reset Page
- Corresponding Tests

Resolves: #23

### CODE REVIEW

**Comment by rishabh1403** [comment]
image
image
image
image
image
image
image

**Comment by ravikp7** [comment]
Looking excellent. Just one question
- Can we get the reset password modal vertically centred on phone view ?
- It seems unreachable to me using one hand.

File: client/src/app.test.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ describe('<App />', () => {
   describe('rendered switch cpomponent', () => {
     it('should render two route component', () => {
```
Comment (ravikp7) [inline client/src/app.test.js]: - This test makes very little sense to me, as we'll be constantly changing this.
- Also, see the description, you seem to forgot to change it from `two` to `five`.

Comment (rishabh1403) [reply client/src/app.test.js]: Yeah, makes sense, maybe we can remove it and add when the pages are done

**Comment by rishabh1403** [comment]
> Looking excellent. Just one question
> 
> * Can we get the reset password modal vertically centred on phone view ?
> * It seems unreachable to me using one hand.

sure

**Comment by rishabh1403** [comment]
image
@ravikp7

File: client/src/components/common/InputFields/EmailField.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import InputField from './InputField';
+
+const EmailField = ({ placeholder }) => (
```
Comment (nimish-gupta) [inline client/src/components/common/InputFields/EmailField.js:5]: Why you need this? you can just create Email field by passing `type=email` to the input field

File: client/src/components/common/InputFields/PasswordField.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import InputField from './InputField';
+
+const PasswordField = ({ placeholder }) => (
```
Comment (nimish-gupta) [inline client/src/components/common/InputFields/PasswordField.js:5]: Why you need this? you can just create Email field by passing `type=password` to the input field

File: client/src/components/common/styledButton/StyledButton.js [hunk 4]
```diff
@@ -4,10 +4,11 @@ import { Button } from 'semantic-ui-react';
 import PropTypes from 'prop-types';
 
 const StyledButton = ({
-  text, url, size, color, link, inverted,
+  text, url, size, color, link, inverted, fluid,
 }) => {
   const buttonProps = {
     text,
+    fluid,
     size: size || 'massive',
```
Comment (nimish-gupta) [inline client/src/components/common/styledButton/StyledButton.js]: set this in defaultProps

File: client/src/pages/signup/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+const SignUp = () => (
+  <div className="signup-form">
+    <Grid textAlign="center" style={{ height: '100%' }} verticalAlign="middle">
+      <Grid.Column style={{ maxWidth: 450 }}>
+        <Header as="h2" color="teal" textAlign="center">
+          Sign Up
+        </Header>
+        <Form size="large">
+          <Segment stacked>
+            <EmailField placeholder="Email" />
```
Comment (nimish-gupta) [inline client/src/pages/signup/SignUp.js:21]: Just use InputField here with `type=email`

**Review by nimish-gupta — CHANGES_REQUESTED** [review]

File: client/src/components/common/InputFields/EmailField.js (hunk 2 above)
Comment (rishabh1403) [reply client/src/components/common/InputFields/EmailField.js:5]: InputField has lot more abstractions...EmailField sets up the icon i want, where I want....
If I directly use InputField I will have to pass all the props, which I know won't change for certain field like email or password or number

File: client/src/components/common/InputFields/PasswordField.js (hunk 3 above)
Comment (rishabh1403) [reply client/src/components/common/InputFields/PasswordField.js:5]: same logic as email field

File: client/src/components/common/styledButton/StyledButton.js (hunk 4 above)
Comment (rishabh1403) [reply client/src/components/common/styledButton/StyledButton.js]: sure

File: client/src/pages/signup/SignUp.js (hunk 5 above)
Comment (rishabh1403) [reply client/src/pages/signup/SignUp.js:21]: same logic as in EmailField abstraction

**Review by nimish-gupta — APPROVED** [review]

**Pull request merged**